enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,210 @@
+---
+title: upgrade-acknowledgment-gate
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@bparees"
+  - "@sdodson"
+  - "@csrwng"
+  - "@relyt0925"
+  - "@jeremyeder"
+approvers:
+  - TBD 
+creation-date: 2021-07-08
+last-updated: 2021-07-14
+status: implementable
+---
+
+# Upgrade Acknowledgment Gate
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently blocks minor level upgrades when overrides are set, a resource deletion is pending, or an operator has set its upgradeable condition to false.
+This enhancement adds a generic upgrade blocking mechanism for blocking minor level upgrades.
+This generic upgrade blocker can consist of one or more items each of which will block a minor level upgrade.
+Each generic upgrade blocker item will be associated with some well-defined task to be completed by an administrator.
+Once the task has been completed the administrator will acknowledge completion and its associated item will no longer block minor level upgrades.
+
+## Motivation
+
+OCP 4.9 uses kubernetes v1.22 which is removing a large number of [APIs][k8s-1.22-removed-apis].
+We suspect that many customer clusters are still using these APIs through helm charts, operators, generic workloads, administrative tools, etc.
+We believe there is no effective way for a Cluster Operator to detect the use of these APIs by external applications.
```

Comment:
4.8 is introduced `apirequestcount` that highlights usage of depreciating APIs in next release. This covers this use-case to a degree but not completely. 

```
[~ {production} (cluster:default)]$ oc get apirequestcounts --as backplane-cluster-admin | grep 1.22
NAME REMOVEDINRELEASE REQUESTINCURRENTHOUR REQUESTSINLAST24H
certificatesigningrequests.v1beta1.certificates.k8s.io                     1.22               0                       373
clusterrolebindings.v1beta1.rbac.authorization.k8s.io                      1.22               32                      2721
customresourcedefinitions.v1beta1.apiextensions.k8s.io                     1.22               0                       488
ingresses.v1beta1.extensions                                               1.22               6                       428
ingresses.v1beta1.networking.k8s.io                                        1.22               0                       0
mutatingwebhookconfigurations.v1beta1.admissionregistration.k8s.io         1.22               0                       253
openshiftapiservers.v1.operator.openshift.io                                                  189                     13223
operators.v1.operators.coreos.com                                                             29                      1822
rolebindings.v1beta1.rbac.authorization.k8s.io                             1.22               0                       0
roles.v1beta1.rbac.authorization.k8s.io                                    1.22               0                       0
validatingwebhookconfigurations.v1beta1.admissionregistration.k8s.io       1.22               0                       170
```

Reply:
I think you got the point based on your mention that it's not complete, but just to be sure I'll clarify. While we can use the new `apirequestcount` in 4.8 to measure and alert on active use of these APIs it doesn't capture atent use, think a CI build pipeline outside of the cluster that hasn't run for months but uses removed APIs to push a critical workload update after it's been upgraded to 4.9. This is a prompt for the admin to go round up all those use cases and work with their userbase and ISVs.

Reply:
The detail in this thread would make a good addition to the body of the doc.

Reply:
I think it is, though perhaps not quite as clearly in line 43.